Mathlib/Analysis/Matrix/Order.lean

```diff
@@ -0,0 +1,260 @@
+    have foo := neg_nonneg.mp <| trace_neg (A - B) ▸ neg_sub A B ▸ h₁.trace_nonneg
+    have : (A - B).trace = 0 := le_antisymm foo h₂.trace_nonneg
+    classical
+    simp_rw [h₂.isHermitian.trace_eq_sum_eigenvalues, ← RCLike.ofReal_sum,
+      RCLike.ofReal_eq_zero, Finset.sum_eq_zero_iff_of_nonneg (s := Finset.univ)
+        (by simpa using h₂.eigenvalues_nonneg), Finset.mem_univ, true_imp_iff] at this
+    exact sub_eq_zero.mp <| funext_iff.eq ▸ h₂.isHermitian.eigenvalues_eq_zero_iff.mp <| this
+
+scoped[MatrixOrder] attribute [instance] Matrix.instPartialOrder
+
+lemma instIsOrderedAddMonoid : IsOrderedAddMonoid (Matrix n n 𝕜) where
+  add_le_add_left _ _ _ _ := by rwa [le_iff, add_sub_add_left_eq_sub]
+
+scoped[MatrixOrder] attribute [instance] Matrix.instIsOrderedAddMonoid
+
+lemma instNonnegSpectrumClass : NonnegSpectrumClass ℝ (Matrix n n 𝕜) where
+  quasispectrum_nonneg_of_nonneg A hA := by
+    classical
+    simp only [quasispectrum_eq_spectrum_union_zero ℝ A, Set.union_singleton, Set.mem_insert_iff,
+      forall_eq_or_imp, le_refl, true_and]
+    intro x hx
+    obtain ⟨i, rfl⟩ := Set.ext_iff.mp
+      hA.posSemidef.1.spectrum_real_eq_range_eigenvalues x |>.mp hx
+    exact (nonneg_iff.mp hA).eigenvalues_nonneg _
+
+scoped[MatrixOrder] attribute [instance] instNonnegSpectrumClass
+
+lemma instStarOrderedRing : StarOrderedRing (Matrix n n 𝕜) :=
+  .of_nonneg_iff' add_le_add_left fun A ↦
+    ⟨fun hA ↦ by
+      classical
+      obtain ⟨X, hX, -, rfl⟩ :=
+        sub_zero A ▸ CFC.exists_sqrt_of_isSelfAdjoint_of_quasispectrumRestricts hA.isHermitian
+          (QuasispectrumRestricts.nnreal_of_nonneg hA.nonneg)
+      exact ⟨X, hX.star_eq.symm ▸ rfl⟩,
+    fun ⟨A, hA⟩ => hA ▸ (posSemidef_conjTranspose_mul_self A).nonneg⟩
+
+scoped[MatrixOrder] attribute [instance] instStarOrderedRing
+
+end PartialOrder
+
+open scoped MatrixOrder
+
+namespace PosSemidef
+
+section sqrtDeprecated
+
+variable [DecidableEq n] {A : Matrix n n 𝕜} (hA : PosSemidef A)
+
+/-- The positive semidefinite square root of a positive semidefinite matrix -/
+@[deprecated CFC.sqrt (since := "2025-09-22")]
+noncomputable def sqrt : Matrix n n 𝕜 :=
+  hA.1.eigenvectorUnitary.1 * diagonal ((↑) ∘ (√·) ∘ hA.1.eigenvalues) *
+  (star hA.1.eigenvectorUnitary : Matrix n n 𝕜)
+
+@[deprecated CFC.sqrt_nonneg (since := "2025-09-22")]
+lemma posSemidef_sqrt : PosSemidef (CFC.sqrt A) := CFC.sqrt_nonneg A |>.posSemidef
+
+include hA
+
+@[deprecated CFC.sq_sqrt (since := "2025-09-22")]
+lemma sq_sqrt : (CFC.sqrt A) ^ 2 = A := CFC.sq_sqrt A
+
+@[deprecated CFC.sqrt_mul_sqrt_self (since := "2025-09-22")]
+lemma sqrt_mul_self : CFC.sqrt A * CFC.sqrt A = A := CFC.sqrt_mul_sqrt_self A
+
+lemma eq_of_sq_eq_sq {B : Matrix n n 𝕜} (hB : PosSemidef B) (hAB : A ^ 2 = B ^ 2) : A = B :=
+  CFC.sqrt_sq A ▸ CFC.sqrt_unique (sq B ▸ hAB.symm)
+
+lemma sq_eq_sq_iff {B : Matrix n n 𝕜} (hB : PosSemidef B) : A ^ 2 = B ^ 2 ↔ A = B :=
+  ⟨eq_of_sq_eq_sq hA hB, fun h => h ▸ rfl⟩
```

Comment:
I still think these should be deprecated. The only one that I think makes sense to keep is `inv_sqrt`, as there is no `⁻¹` in the generic C⋆-algebra case.